src/software/new_geom/convex_polygon_test.cpp

```diff
@@ -49,60 +49,61 @@ TEST(ConvexPolygonConstructorTest, test_construct_from_initializer_list)
-     *      -2       0       2
-     */
-    // Self intersecting polygon, each asterisk on the diagram is a point making up the
-    // polygon
-    EXPECT_THROW(ConvexPolygon({{0.0f, 0.0f},
-                                {0.0f, 2.0f},
-                                {2.0f, 2.0f},
-                                {2.0f, 1.0f},
-                                {0.0f, 1.0f},
-                                {-0.0f, -2.0f},
-                                {-2.0f, -2.0f},
-                                {-2.0f, -1.0f},
-                                {0.0f, -1.0f}}),
-                 std::invalid_argument);
-}
+// Temporarily disabling the following tests until issue #1140 is completed.
+// TEST(ConvexPolygonConstructorTest, test_not_convex)
+//{
+//    /*
+//     *  2            *-------*
+//     *               |       |
+//     *  1            *-------*
+//     *               |
+//     *  0            *
+//     *               |
+//     *  -1   *-------*
+//     *       |       |
+//     *  -2   *-------*
+//     *
+//     *      -2       0       2
+//     */
+//    // Self intersecting polygon, each asterisk on the diagram is a point making up the
+//    // polygon
+//    EXPECT_THROW(ConvexPolygon({{0.0f, 0.0f},
+//                                {0.0f, 2.0f},
+//                                {2.0f, 2.0f},
+//                                {2.0f, 1.0f},
+//                                {0.0f, 1.0f},
+//                                {-0.0f, -2.0f},
+//                                {-2.0f, -2.0f},
+//                                {-2.0f, -1.0f},
+//                                {0.0f, -1.0f}}),
+//                 std::invalid_argument);
+//}
 
-TEST(ConvexPolygonConstructorTest, test_self_intersecting_loop)
-{
-    /*
-     *  3    *------------------------------------------------*
-     *       |                                                |
-     *       |                                                |
-     *  2    |       *--------------------------------*       |
-     *       |             ------           ------            |
-     *       |                  ----      ----                |
-     *  1    |                       ------                   |
-     *       |                ------      ------              |
-     *       |         ------                   -------       |
-     *  0    *------                                   -------*
-     *      -3      -2       -1       0       1       2       3
-     */
-    // Self intersecting polygon, each asterisk on the diagram is a point making up the
-    // polygon
-    EXPECT_THROW(ConvexPolygon({{-3.0f, 0.0f},
-                                {-3.0f, 3.0f},
-                                {3.0f, 3.0f},
-                                {3.0f, 0.0f},
-                                {-2.0f, 2.0f},
-                                {2.0f, 2.0f}}),
-                 std::invalid_argument);
-}
+// TEST(ConvexPolygonConstructorTest, test_self_intersecting_loop)
```

Comment:
Could we leave a TODO here to re-enable this, with  reference to the issue number for the proper `isConvex` check?

Reply:
Done